wave_vision/src/dataset.cpp

```diff
@@ -0,0 +1,255 @@
+#include "wave/vision/dataset.hpp"
+
+namespace wave {
+
+bool TestCamera::update(double dt) {
+    this->dt += dt;
+
+    if (this->dt > (1.0 / this->hz)) {
+        this->dt = 0.0;
+        this->frame++;
+        return true;
+    }
+
+    return false;
+}
+
+int TestCamera::checkFeatures(double dt,
+                              const MatX &features,
+                              const Vec3 &rpy,
+                              const Vec3 &t,
+                              std::vector<std::pair<Vec2, Vec3>> &observed) {
+    Vec3 f_2d, rpy_edn, t_edn;
+    std::pair<Vec2, Vec3> obs;
+    Vec4 f_3d, f_3d_edn;
+    Mat3 R;
+    MatX P;
+
+    // pre-check
+    if (this->update(dt) == false) {
+        return 1;
+    }
+
+    // rotation matrix - convert from nwu to edn then to rotation matrix R
```

Comment:
It matters yes, because in text books when we talk about projection matrices with R and t, they assume you're using the camera coordinate system (EDN: x - right, y - down, z - forward), however, I'm using a Two Wheel Robot to traverse the world, a standard coordinate system for simulators like Gazebo is (NWU: x - forward, y - left, z - up). 

So the tldr; Conversions are necessary if we want to use something like Gazebo simulator to extend this, or we can refactor this and make everything in camera coordinate system (EDN) that would reduce the conversions in this code.